streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -527,12 +527,10 @@ private void runLoop() {
             if (requiresPoll) {
                 requiresPoll = false;
 
-                boolean longPoll = totalNumBuffered == 0;
-
                 ConsumerRecords<byte[], byte[]> records = null;
 
                 try {
-                    records = consumer.poll(longPoll ? this.pollTimeMs : 0);
+                    records = consumer.poll(this.pollTimeMs);
```

Comment:
Just to clarify I think it is not worth having multiple threads within a blocking queue in between for such cases as thread-isolation could be more important than this.